metals/src/main/scala/scala/meta/internal/metals/debug/DebugProxy.scala

```diff
@@ -155,6 +162,29 @@ private[debug] final class DebugProxy(
         }
         .withTimeout(5, TimeUnit.SECONDS)
 
+    case HotCodeReplace(req) =>
+      scribe.info("Hot code replace triggered")
+      compilations
+        .compile(sourceMapper.buildTargets.allBuildTargetIds)
```

Comment:
You can put that into `Compilations` so to not use `buildTargets` from source mapper, but just from Compilations itself.